File: Programmers/level 1/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;


string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    map<char, int> m;
    for (int i=0; i<survey.size(); i++) {
        char disagree=survey[i][0];
        char agree=survey[i][1];
        if (choices[i]==4) continue;
        if (choices[i]>4) {
            //동의가 점수를 얻음
            m[agree]+=(choices[i]-4);
        }
        else if (choices[i]<4) {
            //비동의가 점수를 얻음
            m[disagree]+=(4-choices[i]);
        }
    }
    
    
    answer+=m['R'] >= m['T'] ? 'R':'T';
    answer+=m['C'] >= m['F'] ? 'C':'F';
    answer+=m['J'] >= m['M'] ? 'J':'M';
    answer+=m['A'] >= m['N'] ? 'A':'N';
    
    return answer;
}

File: Programmers/level 1/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    let categories={R:0, T:0, C:0, F:0, J:0, M:0, A:0, N:0};
    
    for (let i=0; i<survey.length; i++) {
        const first = survey[i][0];
        const second = survey[i][1];
        if (choices[i]===4) continue;
        //비동의
        if (choices[i]<4) {
            categories[first]+=(4-choices[i]);
        }
        //동의
        else {
            categories[second]+=(choices[i]-4);
        }
    }
    answer+=categories['R'] >= categories['T'] ? 'R' : 'T';
    answer+=categories['C'] >= categories['F'] ? 'C' : 'F';
    answer+=categories['J'] >= categories['M'] ? 'J' : 'M';
    answer+=categories['A'] >= categories['N'] ? 'A' : 'N';
    
    return answer;
}
